### ISSUE TITLE
Hyperparameter Evolution and Train

### ISSUE THREAD

**Issue by aseprohman** [ultralytics/yolov5#5413]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello @glenn-jocher et all,

why train.py with "--evolve" parameter doesn't create model weights (best.pt or last.pt) after 1 cycle of "evolve" is complete ? is it not possible to find hyperparameter values while constructing model weights ? evolution is expensive computation, I think do this task in paralel is more valuable feature to created :)

at least model weights can generated automatically from the previous of last hyperparameter from "evolve" cycle

regards

**Comment by glenn-jocher** [comment]
@aseprohman fundamentally evolution does not save weights because it is intended to produce a set of hyperparameters rather than a set of weights. A new model can then be trained with these hyperparameters.

Also --evolve includes built in multi-node/VM capabilities for distributed evolution. Problems sometimes occur when multiple agents are reading and writing to evolve.csv simultaneously for example, and this would only be worse for larger files like checkpoints. Lastly, best.pt requires saving one checkpoint every epoch, i.e. up to 300 write operations per generation of fully loaded checkpoints (model, EMA, optimizer gradients), which is not necessary for evolution and adds overhead that can be removed to increase evolution speed slightly.
